src/kvstore/kvstore_dist.h

```diff
@@ -97,6 +101,7 @@ class KVStoreDist : public KVStoreLocal {
         // it may happen for the first time a no-rank-0 worker pull the weight.
         recv_buf = NDArray(grouped_vals[i][0]->shape(), pinned_ctx_);
       }
+      mkl_set_tblob_eager_mode(recv_buf.data());
```

Comment:
Maybe a if case should be added there in case someone does not compile with mkl?